packages/std/src/results/subcall.rs

```diff
@@ -45,6 +47,55 @@ where
     pub reply_on: ReplyOn,
 }
 
+/// This is used for cases when we use ReplyOn::Never and the id doesn't matter
+pub const UNUSED_MSG_ID: u64 = 123456789;
+
+impl<T> SubMsg<T>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    /// new creates a "fire and forget" message with the pre-0.14 semantics
+    pub fn new<M: Into<CosmosMsg<T>>>(msg: M) -> Self {
+        SubMsg {
+            id: UNUSED_MSG_ID,
+            msg: msg.into(),
+            reply_on: ReplyOn::Never,
+            gas_limit: None,
+        }
+    }
+
+    /// create a `SubMsg` that will provide a `reply` with the given id if the message returns `Ok`
+    pub fn reply_on_success<M: Into<CosmosMsg<T>>>(msg: M, id: u64) -> Self {
+        Self::reply_on(msg.into(), id, ReplyOn::Success)
+    }
+
+    /// create a `SubMsg` that will provide a `reply` with the given id if the message returns `Err`
+    pub fn reply_on_error<M: Into<CosmosMsg<T>>>(msg: M, id: u64) -> Self {
+        Self::reply_on(msg.into(), id, ReplyOn::Error)
+    }
+
+    /// create a `SubMsg` that will always provide a `reply` with the given id
+    pub fn reply_always<M: Into<CosmosMsg<T>>>(msg: M, id: u64) -> Self {
+        Self::reply_on(msg.into(), id, ReplyOn::Always)
+    }
+
+    /// add a gas limit to the message. Usage like:
+    ///   SubMsg::reply_always(msg, 1234).with_gas_limit(60_000)
```

Comment:
Let Rust test this docs:

```suggestion
    /// Add a gas limit to the message.
    ///
    /// ## Examples
    ///
    /// ```
    /// # use cosmwasm_std::{coins, BankMsg, CosmosMsg, SubMsg};
    /// # let msg: CosmosMsg = BankMsg::Send { to_address: String::from("you"), amount: coins(1015, "earth") }.into();
    /// let sub_msg = SubMsg::reply_always(msg, 1234).with_gas_limit(60_000);
    /// assert_eq!(sub_msg.gas_limit, Some(60_000));
   ///  ```
```

Reply:
Nice idea. Done